homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -367,4 +377,5 @@ def _get_home(hapid: str) -> Optional[AsyncHome]:
 async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     hap = hass.data[DOMAIN].pop(entry.unique_id)
+    hap.shutdown_listener()
```

Comment:
Is it ok to call this if the listener handler already has run?

Reply:
With the switch back to `hap.async_reset()` in the listener, this is part is only called if the integration is removed. This is not called when the  HA stopps.
So i think it is OK.